install/kubernetes/helm/istio/charts/pilot/templates/clusterrole.yaml

```diff
@@ -35,6 +35,9 @@ rules:
 - apiGroups: [""]
   resources: ["endpoints", "pods", "services", "namespaces", "nodes"]
   verbs: ["get", "list", "watch"]
+- apiGroups: ["discovery.k8s.io"]
+  resources: ["endpointslices"]
```

Comment:
Does this have to be gated by slice enablement?

Reply:
Do you mean will this break old clusters of should we enable this only of they turn on slices?

I think the first part is fine. For the second, I don't think it's worth it since it's an environment variable to enable it right now, maybe if we made it a first class value which I don't think we should do until this is more stable. I don't think giving Endpoints Slice access is too onerous given that is the key part of what pilot does